R/fixed_design_rd.R

```diff
@@ -0,0 +1,110 @@
+#  Copyright (c) 2023 Merck & Co., Inc., Rahway, NJ, USA and its affiliates.
+#  All rights reserved.
+#
+#  This file is part of the gsDesign2 program.
+#
+#  gsDesign2 is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#' Fixed design sample size
+#'
+#' Computes fixed design sample size for risk difference under unstratified design.
+#' Returns a tibble with a basic summary.
+#'
+#' @param alpha One-sided Type I error (strictly between 0 and 1).
+#' @param power Power (`NULL` to compute power or strictly between 0
+#'   and `1 - alpha` otherwise).
+#' @param p_c A numerical value of the control arm rate.
+#' @param p_e A numerical value of the experimental arm rate.
+#' @param rd0 Risk difference under null hypothesis, default is 0.
+#' @param n Sample size. If NULL with power input, the sample size will be
+#' computed to achieve the targeted power
+#' @param ratio Experimental:Control randomization ratio.
+#' @return A table.
+#' @export
+#' @examples
+#' library(dplyr)
+#' # example 1: given power and compute sample size
+#' x <- fixed_design_rd(
+#'   alpha = 0.025, power = 0.9, p_c = .15, p_e = .1,
+#'   rd0 = 0, ratio = 1)
+#' x %>% summary()
+#' # example 2: given sample size and compute power
+#' x <- fixed_design_rd(
+#'   alpha = 0.025, power = NULL, p_c = .15, p_e = .1,
+#'   rd0 = 0, n = 2000, ratio = 1)
+#' x %>% summary()
+fixed_design_rd <- function(alpha = 0.025,
```

Comment:
`rd` has a specific meaning in R used in `man` folder. Could we use a different name? (e.g. `fixed_design_binary` or `fixed_design_risk_difference`.

Reply:
I feel it is okay to use `rd` short for `risk difference`. In the man folder, people clearly know rd is a file format, while in `fixed_design_rd`, users can tell `rd` is not for the file type. Additionally, we have `gs_design_rd` for group sequential design. So let's keep it as `rd` for consistency with `gs_design_rd`.